app/src/main/java/io/homeassistant/companion/android/controls/FanControl.kt

```diff
@@ -53,9 +53,9 @@ class FanControl {
                             entity.entityId,
                             0f,
                             speeds.size.toFloat() - 1,
-                            speeds.indexOf(currentSpeed).toFloat(),
+                            if (entity.state == "on") speeds.indexOf(currentSpeed).toFloat() else 0f,
```

Comment:
Rather than checking for state we should check to see if `speeds.contains(currentSpeed)` because my z-wave fans report speed correctly even when off.